lib/strategy/consul.ex

```diff
@@ -210,18 +210,9 @@ defmodule Cluster.Strategy.Consul do
       |> Keyword.get(:base_url, @default_base_url)
       |> URI.parse()
 
-    case Keyword.get(config, :dc) do
-      nil ->
-        base_url
-
-      dc ->
-        query =
-          (base_url.query || "")
-          |> URI.decode_query(%{"dc" => dc})
-          |> URI.encode_query()
-
-        %{base_url | query: query}
-    end
+    base_url
+    |> maybe_add_query(:dc, config)
+    |> maybe_add_query(:filter, config)
```

Comment:
Maybe (for better performance) we could merge not empty params (from config) and run decode / encode once. Currently if `dc` and `filter` are provided in config we do decoding / encoding twice.